pandas/core/arrays/categorical.py

```diff
@@ -2138,8 +2139,17 @@ def is_dtype_equal(self, other):
         -------
         bool
         """
+        return hash(self.dtype) == hash(other.dtype)
+
+    def is_dtype_equal(self, other) -> bool:
```

Comment:
we should deprecate any is_dtype_equal *methods* on any other Array/Index (not sure if we have any)

Reply:
There's just a CategoricalDtype.is_dtype_equal left.  ill be happy to see that go

Reply:
woops, no, got rid of that a while ago